src/main/java/seedu/address/logic/commands/SortEventCommand.java

```diff
@@ -42,6 +43,23 @@ public int compare(Event o1, Event o2) {
         }
     };
 
+    private static final Comparator<Event> TAG_COMPARATOR = new Comparator<Event>() {
+        @Override
+        public int compare(Event o1, Event o2) {
+            if (o1.getTags().size() != 0 && o2.getTags().size() == 0) {
+                return 1;
+            }
+            if (o1.getTags().size() == 0 && o2.getTags().size() != 0) {
+                return -1;
+            }
+            if (o1.getTags().size() == 0 && o2.getTags().size() == 0) {
+                return o1.getDescription().fullDescription.compareToIgnoreCase(o2.getDescription().fullDescription);
+            }
+            return o1.getTags().iterator().next().tagName
+                    .compareToIgnoreCase(o2.getTags().iterator().next().tagName);
+        }
+    };
```

Comment:
This comparator does not seem to match the description specified in the `indexMessage(Index)` and `MESSAGE_USAGE`. This seems to compare the number of `tag`s instead of the number of participants.

Reply:
The number of participants are counted based on the number of tags because you can add the participants by adding a tag with the participants name. 

What do you think of the above?

Reply:
This doesn't seem to have any indication of the actual number of participants though? It seems to mostly just compare the tagName string-wise. To get the actual number of participants from each tag, I believe you need to use the `model.getPersonsWithTag(Tag)` method to get the actual `Set<Person>`

Reply:
Yes the tags are compared string-wise only when the number of tags (i.e. number of participants) for all events listed are equal. Then the events will be sorted according to the lexicographical order of the participant names. 

If not the sorting order based on the participant number would be:
1. if the events have different participant number, sort according to ascending order. 
2. if a group of events in the list have same participant number, sort based on their description (since they have same number of participants)
3. if all events in the list have same participant number, sort based on the participant name for each event. 

Does this make sense to you? I don't quite get your confusion here. Would appreciate if you could elaborate more.

Reply:
but tags are not the same thing as participants?

Reply:
Okay, will make some changes on the comparison criteria. 

As for your first question, there wasn't much things discussed regarding the tag feature for the events. So Andy and I assumed that names of participants are more or less the main information that the tags will store. Hence I thought comparing the number of tags and the content of tags is acceptable. Would there be a better way to do this?

Reply:
I was under the impression Events would hold a `Set<Person>` (added without any specific tagging) as well as a `Set<Tag>` (of groups of participants identified by the tags). The total number of participants would then be a conjunction of all the participants found from the `Set<Person>` and each `Set<Person>` belonging to each `Tag`.